### ISSUE TITLE
Mantl install issue

### ISSUE THREAD

**Issue by smakam** [CiscoCloud/mantl#1270]
Hi
I am trying to install Mantl in Virtualbox on top of Ubuntu. 

Following is the environment detail:
- Ansible version (`ansible --version`): 1.9.1
- Python version (`python --version`): 2.7.6
- Git commit hash or branch: master

I am facing the following issue, the installation failed and following are logs in the end. Looks like consul install has failed:
```
----- log start ---
TASK: [consul | install nginx template] *************************************** 
failed: [worker-001] => {"changed": true, "cmd": ["consul-cli", "kv-write", "--token=62071f8b-b140-4dc1-a67d-2f60f209269d", "secure/service/nginx/templates/consul", "@/etc/consul/consul.nginx"], "delta": "0:00:01.716756", "end": "2016-03-14 06:38:39.391047", "rc": 1, "start": "2016-03-14 06:38:37.674291", "warnings": []}
stderr: Unexpected response code: 500 (No known Consul servers)

FATAL: all hosts have already failed -- aborting

PLAY RECAP ******************************************************************** 
common | install system utilities ------------------------------------- 203.19s
common | install pip -------------------------------------------------- 144.39s
common | install fail2ban ---------------------------------------------- 95.70s
common | enable EPEL repo ---------------------------------------------- 89.37s
consul | restart consul ------------------------------------------------ 81.82s
common | install distributive ------------------------------------------ 65.51s
consul | deploy tls files ---------------------------------------------- 58.04s
common | update setuptools and pip ------------------------------------- 53.65s
logrotate | copy component logrotate configurations -------------------- 52.10s
consul | install scripts ----------------------------------------------- 50.16s
           to retry, use: --limit @/home/vagrant/sample.retry

control-01                 : ok=91   changed=68   unreachable=0    failed=1   
localhost                  : ok=0    changed=0    unreachable=0    failed=0   
worker-001                 : ok=91   changed=68   unreachable=0    failed=1   

Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
--- Log end ----
```

can you please help? please let me know if any other logs are needed. 

Unrelated question:
In the supported environment, it said Vagrant can be used with Windows for Mantl install? Is that possible?

Thanks 
Sreenivas

**Comment by siddharthist** [comment]
@smakam Windows support is difficult because we use `rsync` shared folders with Vagrant. You might be able to get it working in Cygwin.

**Comment by siddharthist** [comment]
@smakam It looks like some part of the installation failed before that step on the control node. Can you scroll up and see if you can find another failure in the logs?

**Comment by smakam** [comment]
@siddharthist Let me try again since I dont have logs currently. Does logs get stored in some file? Is there a way to run this in a way that ansible retries any failed steps? How do we continue from failed step in case we want to retry?

Thanks
Sreenivas

**Comment by siddharthist** [comment]
@smakam The only way to retry is to run the full playbook again. You can forward the log to a file instead of stdout using 
```
ansible-playbook -e @security.yml  sample.yml >& deployment.log
```

**Comment by smakam** [comment]
@siddharthist This time, I hit this error on "vagrant up":
NOTIFIED: [vault | restart vault] ********************************************* 
failed: [control-01] => {"changed": true, "cmd": ["systemctl", "restart", "vault"], "delta": "0:00:02.633367", "end": "2016-03-15 14:25:23.357158", "rc": 1, "start": "2016-03-15 14:25:20.723791", "warnings": []}
stderr: Job for vault.service failed because a timeout was exceeded. See "systemctl status vault.service" and "journalctl -xe" for details.

I see that the vault.service is running fine in the control node. Not clear why I hit this error. 

When I try to run this command "ansible-playbook -e @security.yml  sample.yml", it does not run in the nodes since inventory file is missing. Do I need to create inventory file manually?

Thanks
Sreenivas

**Comment by smakam** [comment]
@siddharthist For now, I am able to workaround this issue by doing "vagrant reload --provision" and after reload, provisioning worked fine and all the services started fine. 

It will be good to run the playbook manually and also if the playbook can retry in case of errors.

thanks
Sreenivas

**Comment by siddharthist** [comment]
@smakam Sorry, my bad, I didn't see that this was in Vagrant. `vagrant provision` is definitely the best way to rerun the playbook. I'm glad you got it working!
